include/envoy/upstream/thread_local_cluster.h

```diff
@@ -3,6 +3,8 @@
 namespace Envoy {
 namespace Upstream {
 
+class LoadBalancerContext;
```

Comment:
In general only forward declare if it causes a circular include. That might be the case here and elsewhere but just checking.